src/era/rotating.py

```diff
@@ -0,0 +1,106 @@
+"""Create a new directory in base_dir and delete old directories as needed, logging as we go.
+"""
+
+import logging
+import re
+import shutil
+from datetime import datetime
+
+module_logger = logging.getLogger(__name__)
+
+
+class FolderRotator:  # pylint: disable=too-few-public-methods
+    """Create a new directory in base_dir and delete old directories as needed, logging as we go.
+    """
+
+    def __init__(self, base_dir) -> None:
+        self.base_dir = base_dir
+        self._class_logger = logging.getLogger(__name__).getChild(self.__class__.__name__)
+
+    #: Make dir Methods
+    def _build_new_download_dir_path(self, prefix, date_format):
+        today = datetime.today()
+        download_dir = self.base_dir / f'{prefix}{today.strftime(date_format)}'
+        return download_dir
+
+    def _make_new_download_dir(self, download_dir_path, exist_ok):
+        self._class_logger.debug(f'Attempting to create new directory `{download_dir_path}`')
+        try:
+            download_dir_path.mkdir(exist_ok=exist_ok)
+        except FileNotFoundError as error:
```

Comment:
**question**: did you mean for this to be a `FileExistsError` or to add the `parents` [parameter](https://docs.python.org/3/library/pathlib.html#pathlib.Path.mkdir)?

Reply:
I meant to leave `parents` at the default `False` and raise an error that gave a hint more than just a `FileNotFound` error. Personally, I'd rather it fail and warn me than write output to a place I wasn't expecting.

Reply:
Ok, so you are ok with the script exiting here or with a FileExistsError

Reply:
Exactly, this directory needs to exist for the download to work, so it should bomb out with the appropriate notification.